hooks/androidCopyPreferencesPermissions.js

```diff
@@ -0,0 +1,1207 @@
+const fs = require('fs');
+const path = require('path');
+const { ConfigParser } = require('cordova-common');
+const et = require('elementtree');
+const { DOMParser, XMLSerializer } = require('xmldom');
+const { captureRejectionSymbol } = require('events');
+
+module.exports = async function (context) {
+    const projectRoot = context.opts.cordova.project ? context.opts.cordova.project.root : context.opts.projectRoot;
+
+    const configXML = path.join(projectRoot, 'config.xml');
+    const configParser = new ConfigParser(configXML);
+
+    const parser = new DOMParser();
+        
+    // add health connect permissions to AndroidManifest.xml and health_permissions.xml files
+    addHealthConnectPermissionsToXmlFiles(configParser, projectRoot, parser);
+
+    // add background job permissions to AndroidManfiest.xml
+    addBackgroundJobPermissionsToManifest(configParser, projectRoot, parser);
+
+    // copy notification title and content for notificaiton for Foreground Service
+    copyNotificationContent(configParser, projectRoot, parser);
+};
+
+function addHealthConnectPermissionsToXmlFiles(configParser, projectRoot, parser) {
+
+    // Permission groups
+    const fitnessPermissionsRead = [
```

Comment:
maybe this could be dictionaries/objects, instead of arrays with an object that could be more re-usable. 

let's imagine an object like the following, AndroidPermission:
```Typescript
type AndroidPermission = {
      variableName: string, // would be 'HeartRate', 'FitnessVariables', etc) 
      readPermission: string, // it would be the ''android.permission.health.<...>' strings)
      writePermission: string, // it would be the ''android.permission.health.<...>' strings)
      configValue: string // the value we read with <configParser.getPlatformPreference(<name>, 'android')>;
}
```

then a a dictionary of `<String, AndroidPermission>` would be something like:
```Javascript
const permissions = {
        HeartRate: {
              variableName: "HeartRate",
              readPermission: "android.permission.health.READ_HEART_RATE",
              writePermission: "android.permission.health.WRITE_HEART_RATE",
              configValue: ""
        },
        {....}
}
```

and this could be used instead of all these different variables
what do you think?

Reply:
and with this `permissions` dictionary, I believe it's possible to reduce the following ifs, because we can abstract it:
```Javascript
permissions.forEach( (p) => {
        if (p.configValue == "ReadWrite" || p.configValue == "Write") {
              addEntryToManifestAndPermissionsXml(manifestXmlDoc, permissionsXmlDoc, arrayElement, p.writePermission);
        }
});
```